crypto/ml_dsa/ml_dsa_key.c

```diff
@@ -58,10 +53,8 @@ int ossl_ml_dsa_set_prekey(ML_DSA_KEY *key, int prefer_seed, int retain_seed,
     if (seed != NULL
         && (key->seed = OPENSSL_memdup(seed, seed_len)) == NULL)
         goto end;
-    if (prefer_seed >= 0)
-        key->prefer_seed = prefer_seed;
-    if (retain_seed >= 0)
-        key->retain_seed = retain_seed;
+    key->prov_flags |= flags_set;
```

Comment:
this is a strange mismatched notation you are using here..
on/off OR set/clear(or reset)